### ISSUE TITLE
Make it language agnostic

### ISSUE THREAD

**Issue by VictorBjelkholm** [VictorBjelkholm/autochecker#11]
If there is a easy way of detecting which language a project is using, try to detect it and then running the common commands.

Otherwise, use the Dockerfile and the declared CMD.

Fallback to being a NodeJS project.

**Comment by kolodny** [comment]
Supporting a autochecker.json like this would be awesome. you can support arrays for anything that would be a string and either accept an object with `pre`, `override`, and `post` like circle or have a `preinstall`, `install`, and `postinstall` like travis

lightly based off of travis
```json
{
  "images": [
    "mhart/alpine-node:4",
    "mhart/alpine-node:5"
  ],
  "preinstall": ["echo about to install", "echo multiple"],
  "install": "npm install && node mylocalscript.js",
  "test": ["npm run lint", "npm test"]
}
```

lightly based off of circle
```
{
  "images": [
    "mhart/alpine-node:4",
    "mhart/alpine-node:5"
  ],
  "dependencies": {
    "pre": "echo test",
    "override": "npm install"
  }
  "test": ["npm run lint", "npm test"]
}
```

**Comment by VictorBjelkholm** [comment]
I'm not sure this is needed, with a custom Dockerfile, you can achieve the same thing, but not lock in yourself to only using autochecker but also other tools that uses docker.

Anyways, I'll think about it, might make it a bit easier to use and it's not impossible to have both options.

**Comment by VictorBjelkholm** [comment]
This has been implemented now and only #17 is missing to fix, but since the build also pulls, it still works.
